airbyte_cdk/sources/declarative/requesters/query_properties/query_properties.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+
+from dataclasses import InitVar, dataclass
+from typing import Any, Iterable, List, Mapping, Optional, Union
+
+from airbyte_cdk.sources.declarative.requesters.query_properties import (
+    PropertiesFromEndpoint,
+    PropertyChunking,
+)
+from airbyte_cdk.sources.types import Config, StreamSlice
+
+
+@dataclass
+class QueryProperties:
+    """
+    Low-code component that encompasses the behavior to inject additional property values into the outbound API
+    requests. Property values can be defined statically within the manifest or dynamically by making requests
+    to a partner API to retrieve the properties. Query properties also allow for splitting of the total set of
+    properties into smaller chunks to satisfy API restrictions around the total amount of data retrieved
+    """
+
+    property_list: Optional[Union[List[str], PropertiesFromEndpoint]]
+    always_include_properties: Optional[List[str]]
+    property_chunking: Optional[PropertyChunking]
+    config: Config
+    parameters: InitVar[Mapping[str, Any]]
+
+    def get_request_property_chunks(
+        self, stream_slice: Optional[StreamSlice] = None
```

Comment:
In which case does the list of properties depends on the stream_slice?

Reply:
This is for consistency so that the `SimpleRetriever`/`HttpRequester`, etc that is defined to fetch properties from the API endpoint have access to the stream slice object of the current partition. Hubspot specifically doesn't need access to the slice. 

If we don't supply this, then if a future connector were to attempt to use jinja interpolation like we do for query params, this would get resolved to None. So even though we don't current have a use case, I would advocate that we keep this since it doesn't hurt for it to exist

Reply:
I'm ok with that but I would document the reasoning as a docstring for this method so that it's less surprising for other users. Right now, I think my head can wrap the idea that different slices would request for different properties but at the same time, there are weird APIs out there

Reply:
yep i'll add it to the docstring